app/helpers/hosts_helper.rb

```diff
@@ -478,4 +478,28 @@ def inherited_by_default?(field, host)
     return true unless params[:host]
     !params[:host][field]
   end
+
```

Comment:
I like the concept of using the aspects to render tabs, it's quite eloquent.

That said, please check performance - make sure that queries are done as little as possible, for instance, when you bring in an aspect, if you can do that in one query for all aspects, prefer that. I didn't dive too deeply into the code, but it seems like the load_aspects method might do multiple queries where it can have only one, for instance.